### ISSUE TITLE
sokol_app.h: option to not hijack system main function

### ISSUE THREAD

**Issue by floooh** [floooh/sokol#90]
See: https://github.com/floooh/sokol-samples/issues/13#issuecomment-436577552, should be mainly useful for language integrations.

If the user defines a SOKOL_NO_ENTRY before including the implementation, sokol_app.h will not 'hijack' the platform's main entry function, and this must instead be provided by user code.

The user code must call a new ```int sapp_main(sapp_desc*)``` function instead which will take over from there and not return until the application wants to finish, and in this case there shouldn't be a ```sokol_main()``` function to provide the sapp_desc struct (it's basically inverted from the "usual case").

If SOKOL_NO_ENTRY is not defined, sapp_main() will just be an empty function stub (for the reason that SOKOL_NO_ENTRY should only need to be provided for the implementation).

**Comment by code-disaster** [comment]
Ok, that's a more elaborate approach. I like it because it removes the need for a sokol_main() callback.

The hack I use right now is much simpler - effectively I just do:

```
#ifndef SOKOL_ENTRY
#define SOKOL_ENTRY main
#endif
...
int SOKOL_ENTRY(int argc, const** argv) {
}
```

```
#define SOKOL_ENTRY my_main
#define SOKOL_IMPL
#include "sokol_app.h"
```

... or, even more nasty, but works too:

```
#define main my_main
#define SOKOL_IMPL
#include "sokol_app.h"
```

**Comment by floooh** [comment]
Fixed in 3fa7c3ae3793d61aa6c7d29ef5a3848507cde3fe
